### PR TITLE
On prem 2a and stubs

### PR DESCRIPTION
**Pull request by pappasilenus**
This is the first of a set of updates to the on-premises documentation, mapped out in this planning document: https://docs.google.com/document/d/1mfJIwxm1c5timsUTTMIMvfOaIZuJS8sB3lnsrxXfYuM/edit

This represents document 2.a only, with stubs for all the other documents.  2.b, in particular, is somewhat filled out but needs more work and will be worked on next.

### CODE REVIEW

File: doc/deployment/deploy_custom.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+Please see the [introduction to on-premises deployment](./on_premises.html) for steps you need to take prior to creating a Pachydemerm deployment manifest.
+
+## Creating a Pachyderm manifest
+
+Please see the [introduction to on-premises deployment](./on_premises.html) for an explanation of the differences among static persistent volumes, StatefulSets and StatefulSets with StorageClasses, as well as the meanings of the variables, like  `PVC_STORAGE_SIZE` and `OS_ENDPOINT`, used below.
+
+### Configuring with a static persistent volume
+The command you'll want to run is 
+```sh
+$ pachctl deploy custom --persistent-disk aws --object-store s3
```
[the remark this answers is not in the document]
Comment (JoeyZwicker) [reply doc/deployment/deploy_custom.md]: This is a bit confusing. It'd be nice if we had a better way

File: doc/deployment/on_premises.md [hunk 2]
```diff
@@ -1,33 +1,181 @@
 # On Premises
 
-Pachyderm is built on [Kubernetes](https://kubernetes.io/) and can be backed by an object store of your choice. As such, Pachyderm can run on any on premise platforms/frameworks that support Kubernetes, a persistent disk/volume, and an object store.
+This document is broken down into the following sections, available at the links below
+
+- [Introduction to on-premises deployments](#introduction) takes you through what you need to know about Kubernetes, persistent volumes, object stores and best practices.  That's this page.
+- [Customizing your Pachyderm deployment for on-premises use](./deploy_custom.html) details the various options of the `pachctl deploy custom ...` command for an on-premises deployment.
+- [Single-node Pachyderm deployment](./single-node.html) is the document you should read when deploying Pachyderm for personal, low-volume usage.
```
Comment (JoeyZwicker) [inline doc/deployment/on_premises.md]: Single-node here is distinctly different than deploy local, right? Probably worth clarifying. Maybe just add "in a persistent non-dev way" at the end of that bullet to clarify?

File: doc/deployment/on_premises.md [hunk 3]
```diff
@@ -1,33 +1,181 @@
...
+
+## Best practices
+### Infrastructure as code
+We highly encourage you to apply the best practices used in developing software to managing the deployment process.
+1. Create scripts that automate as much of your processes as possible and keep them under version control.
+1. Keep copies of all artifacts, such as manifests, produced by those scripts and keep those under version control.
+1. Document your practices in the code and outside it.
+### Infrastructure in general
+Be sure that you design your Kubernetes infrastructure in accordance with recommended guidelines.
+Don't mix on-premises Kubernetes and cloud-based storage.
```
Comment (JoeyZwicker) [inline doc/deployment/on_premises.md]: why not? Just performance reasons or do we actually not support this?

File: doc/deployment/on_premises.md [hunk 4]
```diff
@@ -1,33 +1,181 @@
...
+   
+### Deploying an object store
+
+#### Object store: what's it for?
+An object store is used by Pachyderm's `pachd` for storing all your data. 
+The object store you use must be accessible via a low-latency, high-bandwidth connection like [Gigabit](https://en.wikipedia.org/wiki/Gigabit_Ethernet)  or [10G Ethernet](https://en.wikipedia.org/wiki/10_Gigabit_Ethernet).
+
+For an on-premises deployment, 
+it's not advisable to use a cloud-based storage mechanism.
+Don't deploy an on-premises Pachyderm cluster against cloud-based object stores such as S3 from [AWS](https://pachyderm.readthedocs.io/en/latest/deployment/amazon_web_services.html), GCS from [Google Cloud Platform](https://pachyderm.readthedocs.io/en/latest/deployment/google_cloud_platform.html), Azure Blob Storage from  [Azure](https://pachyderm.readthedocs.io/en/latest/deployment/azure.html).
```
Comment (JoeyZwicker) [inline doc/deployment/on_premises.md]: Maybe clarify why not? Performance?

File: doc/deployment/deploy_custom.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pappasilenus) [reply doc/deployment/deploy_custom.md]: That's something I'll clarify in the next update.  In short, you need to specify one of those and then edit it.

File: doc/deployment/on_premises.md (hunk 3 above)
Comment (pappasilenus) [reply doc/deployment/on_premises.md]: It could be done.Performance will make it unusable.

File: doc/deployment/on_premises.md (hunk 4 above)
Comment (pappasilenus) [reply doc/deployment/on_premises.md]: I could clarify. It's a little clearer further down where I specify the performance requirements for on-prem storage deployments (gig or 10G ethernet or better).  

There are hybrid cloud environments where this is feasible, but the reason it's feasible is that there's a lot of peering-style networking going on, along with on-prem caching.

File: doc/deployment/on_premises.md (hunk 2 above)
Comment (pappasilenus) [reply doc/deployment/on_premises.md]: That's clarified in the actual single-node document, I thought. I specify that `deploy local` is only good for testing and evaluation.

**Pull request merged**